servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TestExecutor.java

```diff
@@ -418,7 +418,8 @@ public void clear() {
 
         @Override
         public AsyncContextMap copy() {
-            throw invalidAccess();
+            // Create a new non-broken empty map as this is most likely a subscribe()
+            return new CopyOnWriteAsyncContextMap();
```

Comment:
here is an alternative which uses the public APIs:

```java
AsyncContextMap map = AsyncContext.current().copy();
map.clear();
return map;
```

Reply:
In most cases the current map would be this invalid one so copying it would be just recursively calling the same method.

Reply:
I think the only problem with this approach is that we are 'enforcing' one implementation, whereas in the prod code this can be in the future a CHM.

Reply:
There are two alternative directions to take. Simple and fancy:

- Simple: Have `copy()` return `this` just as `NoopAsyncContextMap` does.
- Fancy: Add a `newContextMap()` method to the `AsyncContextProvider` interface with suitable implementation to abstract the map creation.

Opinions?

Reply:
I opted to "simple" in the PR.